### PR TITLE
Add CSS property whitelisting to the sanitizer

### LINKED ISSUE #230: Sanitizer does not support CSS properties
Whitelist class only supports tags and attributes, but not CSS style like other HTML sanitizer library for web, e.g [DOMPurify](https://github.com/cure53/DOMPurify), [HtmlSanitizer](https://github.com/jitbit/HtmlSanitizer). Is it possible to add this functionality?

### PR DESCRIPTION
**Pull request by rursache**
Closes #230

## Summary
- add `Whitelist.addCSSProperties(_:_:)` and `removeCSSProperties(_:_:)` for tag-scoped inline CSS property whitelisting, including `:all` support
- filter `style` attributes during sanitization so only allowed declarations survive, and drop the attribute when no safe declarations remain
- harden inline CSS sanitization by stripping CSS comments and rejecting common unsafe vectors such as `url(...)`, `expression(...)`, `@import`, `behavior`, and `-moz-binding`
- document the new API in the README

## Tests
- add cleaner tests for property filtering, dropping empty styles, `:all` behavior, case-insensitive property names, parsing quoted semicolons/functions, comment-aware parsing, and unsafe CSS values/properties
- `swift test --filter CleanerTest`
- `swift test`

## Notes
- the new CSS filtering only applies when CSS properties are explicitly configured for a `style` attribute; existing sanitizer behavior is unchanged otherwise

### CODE REVIEW

File: Sources/Whitelist.swift [hunk 1]
```diff
@@ -618,6 +711,227 @@ import Foundation
...
+        }
+
+        if let declaration = parseStyleDeclaration(buffer) {
+            declarations.append(declaration)
+        }
+
+        return declarations
+    }
+
+    private func stripCSSComments(_ style: String) -> String {
```
Comment (aehlke) [inline Sources/Whitelist.swift]: of course profiling will show what deserves optimization, but this kind of method is generally good to defensively optimize by operating on utf8 bytes directly. I can help spin up on how to do that if you like
